pkg/controller/manager/sync_action.go

```diff
@@ -201,14 +204,27 @@ func syncRepairTasks(clusterID string, cluster *v1.ScyllaCluster, syncer stateCa
 	return actions, nil
 }
 
+type startDateGetterSetter interface {
+	GetStartDate() string
+	SetStartDate(sd string)
+}
+
+func taskChanged(spec, managerTask startDateGetterSetter) bool {
+	// Keep special "now" value evaluated on task creation.
+	if strings.HasPrefix(spec.GetStartDate(), "now") {
+		spec.SetStartDate(managerTask.GetStartDate())
```

Comment:
this operation is quite hidden for the caller given the function name and it seems to transitively write into a shared informer

Reply:
good catch, fixed into writing into a copy

Reply:
@zimnx as rewriting the startDate doesn't really fit into the scope of `taskChanged`, can you pls extract it to a defaulting function, say `evaluateDates`, and call it right after the `DeepCopy` in `syncRepairTasks` and `syncBackupTasks`?